src/phase_utils.py

```diff
@@ -1,10 +1,9 @@
 from enum import Enum
+from pathlib import Path
+import json
+import numpy as np
 
 
-###############################################################################
-# Utilitiy
-###############################################################################
-
```

Comment:
Personally i think the dividers are visually pleasing. 
But existing ones could be replaced by, for example:

```
###
# Classes
###
```

```
###
# Functions
###
```

Or

Split utility into 2 files. 
If this approach is taken, then make sure all code works.